src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+import { Box, Typography, useTheme, Link } from '@mui/material';
+import React, { useState } from 'react';
+
+import FacebookStatic from './assets/Facebook.svg?url';
+import GithubStatic from './assets/Github.svg?url';
+import GoogleStatic from './assets/Google.svg?url';
+import {
+  getPageContainerStyles,
+  getFormStyles,
+  getLogoStyles,
+  getInputStyles,
+  getButtonStyles,
+  haveAccountTextStyles,
+  loginLinkStyles,
+  signupWithTextStyles,
+  getSocialIconStyles,
+  legalTextStyles,
+  legalLinkStyles,
+} from './SignUpPage.styles';
+
+import type { SignUpPageProps } from './SignUpPage.types';
+
+const FacebookIcon: string | undefined = FacebookStatic ?? undefined;
+const GithubIcon: string | undefined = GithubStatic ?? undefined;
+const GoogleIcon: string | undefined = GoogleStatic ?? undefined;
+
+type LogoFallbackProps = {
+  orientation?: string;
+  width?: string | number;
+  height?: string | number;
+  alt?: string;
+};
+
+type FormInputFallbackProps = {
+  label?: string;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  errorMessage?: string;
+  sx?: Record<string, unknown>;
+  type?: string;
+};
+
+type PrimaryButtonFallbackProps = {
+  children?: React.ReactNode;
+  onClick?: (e: React.FormEvent) => void;
+  disabled?: boolean;
+  loading?: boolean;
+  sx?: Record<string, unknown>;
+};
+
+const Logo: React.ComponentType<LogoFallbackProps> = () => <div data-testid="logo-fallback" />;
+
+const FormInput: React.FC<FormInputFallbackProps> = (props) => (
+  <>
+    <Box
+      data-testid={`form-input-fallback-${props.placeholder ?? 'input'}`}
+      component="div"
+      sx={(props.sx as unknown as Record<string, unknown>) ?? { width: '100%', height: '40px' }}
+      aria-hidden="true"
+    />
+    <input
+      placeholder={props.placeholder}
+      defaultValue={props.value}
+      onChange={props.onChange}
+      type={props.type}
+      style={{
+        position: 'absolute',
```

Comment:
The native <input> is visually positioned off-screen (left: '-9999px') to be accessible while the visible control is a non-semantic Box. Off-screen positioning can cause focus and screen-reader issues. Off-screen inputs should only be used carefully (e.g. skip links). For form inputs, the semantic control should be visible and labelled. This implementation will be confusing for keyboard and screen reader users and is likely to fail accessibility audits.

**Sugerencia:**
```typescript
Render the real <input> where users expect it; use proper labels. If the goal is to keep layout stable, style the input using MUI components and theme values instead of hiding it offscreen.
```